test/powershell/Host/ConsoleHost.Tests.ps1

```diff
@@ -163,6 +163,20 @@ Describe "ConsoleHost unit tests" -tags "Feature" {
             $actual | Should Be $expected
         }
 
+        It "-Version should return the engine version" {
+            $currentVersion = "powershell " + $PSVersionTable.GitCommitId.ToString()
+            $observed = & $powershell -version
+            $observed | should be $currentVersion
+        }
+
+        It "-Version should ignore other parameters" {
+            $currentVersion = "powershell " + $PSVersionTable.GitCommitId.ToString()
+            $observed = & $powershell -version -command get-date
```

Comment:
it would report the version and exit - I can add a test for that explicitly if you like. Rather than testing all of the permutations for the parameters, I chose a representative single case

---
In reply to: [100405816](https://github.com/PowerShell/PowerShell/pull/3115#discussion_r100405816) [](ancestors = 100405816)